server/src/main/java/keywhiz/KeywhizService.java

```diff
@@ -157,7 +159,8 @@ public void addSecretGenerator(String name, SecretGenerator<?> secretGenerator)
     jersey.register(injector.getInstance(ClientCertificateFilter.class));
 
     logger.debug("Registering response filters");
-    jersey.register(injector.getInstance(SecurityHeadersFilter.class));
+    environment.servlets().addFilter("security-headers-filter", injector.getInstance(SecurityHeadersFilter.class))
+        .addMappingForUrlPatterns(EnumSet.of(DispatcherType.REQUEST), true, "/*");
```

Comment:
Should this be set for ERROR dispatch types too? Could you move this down under "Registering servlet filters" and use inline comments like XsrfServletFilter does?